### PR TITLE
feat: Add new config parameter `private_key`

### PR DESCRIPTION
**Pull request by haleemur**
# Description

Add a config parameter `private_key`. This allows a user of target-snowflake to specify the `private_key` directly.

This is useful in cases where it is difficult to save the private_key to a file but easy to pass in an environment variable (e.g. a container)

### CODE REVIEW

File: target_snowflake/connector.py [hunk 1]
```diff
@@ -124,6 +127,46 @@ def _convert_type(sql_type):  # noqa: ANN205, ANN001
...
+        )
+
+        return p_key.private_bytes(
+            encoding=serialization.Encoding.DER,
+            format=serialization.PrivateFormat.PKCS8,
+            encryption_algorithm=serialization.NoEncryption(),
+        )
+
+    @cached_property
+    def auth_method(self):
```
Comment (edgarrmondragon) [inline target_snowflake/connector.py:153]: I didn't comment a similar thing for tap-snowflake, but wdyt about making this return an enum member (e.g. `SnowflakeAuthMethod`) so using `self.auth_method` is slight type-safer?

Comment (haleemur) [reply target_snowflake/connector.py]: sure thing. please see updated.

File: target_snowflake/connector.py [hunk 2]
```diff
@@ -62,6 +66,15 @@ def prepare_part(self, part: str) -> str:
         return self.dialect.identifier_preparer.quote(part)
 
 
+class SnowflakeAuthMethod(Enum):
+    """Supported methods to authenticate to snowflake"""
+
+    BROWSER = 1
+    PASSWORD = 2
+    PRIVATE_KEY = 3
+    PRIVATE_KEY_PATH = 4
```
[the remark this answers is not in the document]
Comment (haleemur) [reply target_snowflake/connector.py]: done.

**Review by edgarrmondragon — APPROVED** [review]
Awesome, thanks @haleemur!

**Pull request merged**